docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -43,6 +40,56 @@ We plan to include the following in future versions of the guidelines:
 
 - *Provider*: A callable step in a workflow writing with Sciline.
 
+## C: Convention
+
+### C.1: Use common names and types
+
+**Reason**
+Helps with sticking to established practices and working across packages.
+
+**Table**
+Names use glob syntax, i.e., '*Filename' is any string that ends in 'Filename'.
+
+| Name                          | Type        | Description                                                               |
+|-------------------------------|-------------|---------------------------------------------------------------------------|
+| --- **Files** ---             |             |                                                                           |
+| Filename \| *Filename         | str         | Simple name of a file, must be processed into FilePath                    |
+| FilePath \| *FilePath         | Path        | Concrete path to a file on the host filesystem, ideally absolute          |
+| --- **Run IDs** ---           |             |                                                                           |
+| SampleRun, BackgroundRun, ... | Any         | Identifier for a run                                                      |
+| RunType                       | TypeVar     | Constrained to the run types used by the package, see above               |
+| RunTitle                      | str         | Extracted from NeXus or provided by user, can be used to find files       |
+| --- **Monitors** ---          |             |                                                                           |
+| IncidentMonitor \| *Monitor   | Any         | Identifier for a monitor                                                  |
+| MonitorType                   | TypeVar     | Constrained to the monitor types used by the package, see above           |
```

Comment:
Here we assume that a workflow necessarily wants/needs to use generics for runs types and monitor types. Is that a good assumption? Or should we consider formulating it as "if using generics for X, use this convention"?

Should we maybe split `C.1` to discuss naming for generics (and their typevars) separately?

Reply:
> Here we assume that a workflow necessarily wants/needs to use generics for runs types and monitor types. Is that a good assumption? Or should we consider formulating it as "if using generics for X, use this convention"?

I don't understand what you are saying. When we don't use generics, the SampleRun, ... and RunType simply don't exist. This table doesn't mandate using all entries, it's supposed to tell us how to name them if and when we need them.

> Should we maybe split C.1 to discuss naming for generics (and their typevars) separately?

Why? How would you structure that table?

Reply:
> I don't understand what you are saying. When we don't use generics, the SampleRun, ... and RunType simply don't exist.

Say I want to write a workflow that process a sample run and a vanadium run, without use of generics. Which names do I use?

I am suggesting to make two guidelines (each with their own table), or have a separate guideline specifying conventions for naming typevars and identifiers.

Reply:
If the reader is not familiar with using generic providers in a workflow then the current table is quite unclear. Please make a separate guidelines. We do not need to make an explicit list.